socorro/external/es/super_search_fields.py

```diff
@@ -136,26 +136,53 @@ def add_field_to_properties(properties, namespaces, field):
                 field,
             )
 
+        def is_doc_values_friendly(field_mapping):
+            field_type = field_mapping.get('type')
+
+            # No clue what type this is--probably false
+            if not field_type:
+                return False
+
+            # boolean fields don't work with doc_values=True
+            if field_type == 'boolean':
+                return False
+
+            # object fields shouldn't get doc_values=True
+            if field_type == 'object':
+                return False
+
+            # analyzed string fields don't work with doc_values=True
+            if field_type == 'string' and field_mapping.get('index') != 'not_analyzed':
+                return False
+
+            # Everything is fine! Yay!
+            return True
+
+        def add_doc_values(props):
+            """Add "doc_values": True to properties
+
+            NOTE(willkg): Elasticsearch 2.0+ does this automatically, so we
+            can nix this.
+
+            """
+            if is_doc_values_friendly(props):
+                props['doc_values'] = True
+
+            # Handle subfields
+            if props.get('fields'):
+                for field in props.get('fields', {}).values():
+                    add_doc_values(field)
+
+            # Handle objects with nested properties
+            if props.get('properties'):
+                for field in props['properties'].values():
+                    add_doc_values(field)
```

Comment:
These should be extracted. It'd be a lot easier to test and verify them if they weren't defined in the method they're used.